projects/plugins/jetpack/modules/sitemaps/sitemap-librarian.php

```diff
@@ -108,6 +113,25 @@ public function store_sitemap_data( $index, $type, $contents, $timestamp ) {
 		}
 	}
 
+	/**
+	 * Get the current sitemap post ID.
+	 *
+	 * @param string $name The name of the sitemap.
+	 * @param string $type The type of the sitemap.
+	 * @return int|null The post ID if it exists, null otherwise.
+	 */
+	private function get_current_sitemap_post_id( $name, $type ) {
+		$args = array(
+			'post_type'      => $type,
+			'post_status'    => 'draft',
+			'posts_per_page' => 1,
+			'title'          => $name,
+			'fields'         => 'ids',
+		);
+
+		$query = new WP_Query( $args );
+		return $query->posts ? $query->posts[0] : null;
```

Comment:
I have one question about this. Should we use the `WP_Query` for checking if it has posts instead, like `$query->has_posts()`?

Reply:
I think this is simple enough with the have_posts check. Looking at the source for have_posts, I don't see any benefits to use that than a simple check of the posts property.